packages/server/src/adapters/fetch/content-type/json/index.ts

```diff
@@ -0,0 +1,96 @@
+// @trpc/server
+import { TRPCError } from '../../../../@trpc/server';
+import type {
+  AnyRouter,
+  CombinedDataTransformer,
+} from '../../../../@trpc/server';
+import type { BaseContentTypeHandler } from '../../../../@trpc/server/http';
+import type { FetchHandlerRequestOptions } from '../../types';
+
+export interface FetchHTTPContentTypeHandler<TRouter extends AnyRouter>
+  extends BaseContentTypeHandler<
+    FetchHandlerRequestOptions<TRouter> & {
+      url: URL;
+    }
+  > {}
+
+export const getFetchHTTPJSONContentTypeHandler: <
+  TRouter extends AnyRouter,
+>() => FetchHTTPContentTypeHandler<TRouter> = () => ({
+  name: 'fetch-json',
+  isMatch(opts) {
+    return !!opts.req.headers
+      .get('content-type')
+      ?.startsWith('application/json');
+  },
+  getInputs: async (opts, info) => {
+    async function getRawProcedureInputOrThrow() {
+      const { req } = opts;
+
+      try {
```

Comment:
I think we can avoid big try/catch blocks but dis is ok

Reply:
This method may have earned some refactoring either now or in the future, I mostly copied it into place as existing code and made some tweaks

It's actually quite battle hardened though since I didn't write it